packages/api/src/WithIntegrationHelpers.tsx

```diff
@@ -0,0 +1,352 @@
+        draft.flows = [];
+      }
+      if (!draft.flows[flow]) {
+        draft.flows[flow] = {
+          id: key(),
+          name: '',
+          steps: [],
+        };
+      }
+      if (!draft.flows[flow].steps) {
+        draft.flows[flow].steps = [];
+      }
+      const step: Step = {
+        action,
+        configuredProperties,
+        connection,
+        id: draft.flows[flow].id,
+      };
+      if (actionDescriptor) {
+        step.action!.descriptor = actionDescriptor;
+      }
+      step.stepKind = 'endpoint';
+      draft.flows[flow].steps![position] = step;
+    });
+  }
+  public async updateOrAddConnection(
+    integration: Integration,
+    connection: Connection,
+    action: Action,
+    flow: number,
+    position: number,
+    configuredProperties: any
+  ): Promise<Integration> {
+    const actionDescriptor = await this.getActionDescriptor(
+      connection.id!,
+      action.id!,
+      configuredProperties
+    );
+    return produce(integration, draft => {
+      if (!draft.flows) {
+        draft.flows = [];
+      }
+      if (!draft.flows[flow]) {
+        draft.flows[flow] = {
+          id: key(),
+          name: '',
+          steps: [],
+        };
+      }
+      if (!draft.flows[flow].steps) {
+        draft.flows[flow].steps = [];
+      }
+      const step: Step = {
+        action,
+        configuredProperties,
+        connection,
+        id: draft.flows[flow].id,
+      };
+      if (actionDescriptor) {
+        step.action!.descriptor = actionDescriptor;
+      }
+      step.stepKind = 'endpoint';
+      if (draft.flows[flow].steps![position]) {
+        draft.flows[flow].steps![position] = step;
+      } else {
+        draft.flows[flow].steps!.splice(position, 0, step);
+        draft.tags = Array.from(
+          new Set([...(draft.tags || []), connection.id!])
+        );
+      }
+    });
+  }
+
+  public getEmptyIntegration(): Integration {
+    return NEW_INTEGRATION;
+  }
+
+  public async saveIntegration(integration: Integration): Promise<Integration> {
+    const response = await callFetch({
+      body: integration,
+      headers: this.props.headers,
+      method: integration.id ? 'PUT' : 'POST',
+      url: integration.id
+        ? `${this.props.apiUri}/integrations/${integration.id}`
+        : `${this.props.apiUri}/integrations`,
+    });
+    if (!response.ok) {
+      throw new Error(response.statusText);
+    }
+    return !integration.id
+      ? ((await response.json()) as Integration)
+      : Promise.resolve(integration);
+  }
+
+  public setName(integration: Integration, name: string): Integration {
+    return produce(integration, nextIntegration => {
+      nextIntegration.name = name;
+    });
+  }
+
+  public makeLocalStorageId(id: string) {
+    return `iec-${id}`;
+  }
+
+  public async createDraft(value: Integration) {
+    // TODO: this should be handled by the BE
+    if (!value.id) {
+      throw new Error("Integration has no id, can't create a draft");
```

Comment:
We *could* create this ID easily enough [this here](https://github.com/syndesisio/syndesis/blob/master/app/ui/src/app/platform/helpers/key.ts) is how it's done in the angular codebase.

Reply:
Rofl, and here I find the file in this PR :-)

Reply:
Actually, that file and the draft methods are leftovers that I forgot to remove! 

So the idea to keep the integration state - before me realizing that the history could do that for me - was to call an API to store the draft and retrieve it when needed. These helpers were supposed to do that and since we have no such API the plan was to fall back on the localStorage. 

I'll do some cleanup and remove the unused stuff.